core/src/main/java/io/questdb/jit/CompiledFilterIRSerializer.java

```diff
@@ -1029,6 +1034,37 @@ private void serializeIn() throws SqlException {
         }
     }
 
+    private void serializeInTimestampRange(int position) throws SqlException {
+        predicateContext.currentInSerialization = true;
+
+        final CharSequence token = predicateContext.inOperationNode.rhs.token;
+        final CharSequence intervalEx = token == null || SqlKeywords.isNullKeyword(token) ? null : GenericLexer.unquote(token);
+
+        final LongList intervals = predicateContext.inIntervals;
+        IntervalUtils.parseAndApplyIntervalEx(intervalEx, intervals, position);
+
+        final PostOrderTreeTraversalAlgo traverseAlgo = new PostOrderTreeTraversalAlgo();
```

Comment:
nit: would be great to reuse this object across JIT compilations.

Reply:
Done, also added a new test `testInAndInTimestamp` to ensure the single `PostOrderTreeTraversalAlgo` handles multiple `in` expressions correctly without one stepping on the other.